h2o-hive/src/main/java/water/hive/DelegationTokenRefresher.java

```diff
@@ -28,11 +28,18 @@
   public static final String H2O_HIVE_JDBC_URL_PATTERN = "h2o.hive.jdbc.urlPattern";
   public static final String H2O_HIVE_HOST = "h2o.hive.jdbc.host";
   public static final String H2O_HIVE_PRINCIPAL = "h2o.hive.principal";
+  public static final String H2O_HIVE_TOKEN = "h2o.hive.token";
 
   public static void setup(Configuration conf, String tmpDir) throws IOException {
     if (!HiveTokenGenerator.isHiveDriverPresent()) {
       return;
     }
+    String token = conf.get(H2O_HIVE_TOKEN);
+    if (token != null) {
+      log("Adding credentials from property", null);
+      Credentials creds = HiveTokenGenerator.tokenToCredentials(token);
+      UserGroupInformation.getCurrentUser().addCredentials(creds);
+    }
     String authUser = conf.get(H2O_AUTH_USER);
     String authPrincipal = conf.get(H2O_AUTH_PRINCIPAL);
     String authKeytab = conf.get(H2O_AUTH_KEYTAB);
```

Comment:
Does `H2O_AUTH_KEYTAB` still need to be passed into setup method?

Reply:
no , you can use use H2O_HIVE_TOKEN H2O_HIVE_JDBC_URL_PATTERN H2O_HIVE_PRINCIPAL H2O_HIVE_HOST